framework/src/interfaces/MortarExecutorInterface.C

```diff
@@ -0,0 +1,25 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "MortarExecutorInterface.h"
+#include "FEProblem.h"
+#include "MortarData.h"
+
+MortarExecutorInterface::MortarExecutorInterface(FEProblemBase & fe_problem)
+  : _mortar_data(fe_problem.mortarData())
+{
+  _mortar_data.notifyWhenMortarSetup(this);
+}
+
+MortarExecutorInterface::MortarExecutorInterface(MortarExecutorInterface && other)
+  : _mortar_data(other._mortar_data)
+{
+  _mortar_data.dontNotifyWhenMortarSetup(&other);
+  _mortar_data.notifyWhenMortarSetup(this);
+}
```

Comment:
This part I don't fully grasp. How is it `MortarExecutorInterface` deciding what to notify and what to denotify? If I understand well, notification implies that a `MortarExecutorInterface` needs to run `mortarSetup()` to find out dependencies.

Reply:
If a move constructor gets called, that means the constructor argument (`other` in is case) is going to get destroyed after move construction of the new object. So if you don't remove the `other` `MortarExecutorInterface` from the container of objects that `MortarData` will iterate over and call `mortarSetup` on after the mortar mesh has been created, then you will get a segmentation fault because you will calling `mortarSetup` through a pointer to an object that has been destroyed.